src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -404,15 +410,15 @@ public bool Equals(TypeWithAnnotations other, TypeCompareKind comparison)
 
         internal sealed class EqualsComparer : EqualityComparer<TypeWithAnnotations>
         {
-            internal static readonly EqualsComparer ConsiderEverythingComparer = new EqualsComparer(TypeCompareKind.ConsiderEverything);
-            internal static readonly EqualsComparer IgnoreNullableModifiersForReferenceTypesComparer = new EqualsComparer(TypeCompareKind.IgnoreNullableModifiersForReferenceTypes);
-            internal static readonly EqualsComparer UnknownNullableModifierMatchesAnyComparer = new EqualsComparer(TypeCompareKind.ObliviousNullableModifierMatchesAny);
+            internal static readonly EqualsComparer ConsiderEverythingComparer = new EqualsComparer(TypeCompareKind.ConsiderEverything, isValueTypeOverrideOpt: null);
 
             private readonly TypeCompareKind _compareKind;
+            private readonly SmallDictionary<TypeParameterSymbol, bool> _isValueTypeOverrideOpt;
```

Comment:
>SmallDictionary [](start = 29, length = 15)

It's unfortunate this is a mutable dictionary. Should all of the `Equals()` infrastructure use `ImmutableDictionary<TypeParameterSymbol, bool>` instead?